### ISSUE TITLE
LogStreamTest.shouldIncreasePositionOnRestart flaky

### ISSUE THREAD

**Issue by pihme** [zeebe-io/zeebe#4543]
**Summary**

- How often does the test fail? - top3 of flaky tests CW 21, CW22
- Does it block your work? - no
- Do we suspect that it is a real failure? - unknown

**Failures**

https://ci.zeebe.camunda.cloud/job/zeebe-io/job/zeebe/job/4340-do-not-merge-monitor-build-health-branch/671

<details><summary>Example assertion failure</summary>
<pre>
Error Message

Expecting:
 <-1L>
to be greater than:
 <4294967488L> 

Stacktrace

java.lang.AssertionError: 

Expecting:
 <-1L>
to be greater than:
 <4294967488L> 
	at io.zeebe.logstreams.log.LogStreamTest.shouldIncreasePositionOnRestart(LogStreamTest.java:113)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeLazy(JUnitCoreWrapper.java:119)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:87)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:377)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:138)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:465)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:451)
</pre>
</details>

**Hypotheses**

**Logs**

> If possible, provide more context here, e.g. standard output logs, link to build, etc.

<details><summary>Logs</summary>
<pre>
12:14:31.066 [] [main] INFO  io.zeebe.test - Test finished: shouldBuildLogStream(io.zeebe.logstreams.log.LogStreamTest)
12:14:31.067 [] [main] INFO  io.zeebe.test - Test started: shouldIncreasePositionOnRestart(io.zeebe.logstreams.log.LogStreamTest)
12:14:31.659 [Broker-0-LogStream-0] [-zb-actors-4] DEBUG io.zeebe.logstreams - Configured log appender back pressure at partition 0 as AppenderVegasCfg{initialLimit=1024, maxConcurrency=32768, alphaLimit=0.7, betaLimit=0.95}. Window limiting is disabled
12:14:31.767 [Broker-0-LogStream-0] [-zb-actors-3] INFO  io.zeebe.logstreams - Close appender for log stream 0
12:14:31.769 [0-write-buffer] [-zb-actors-3] DEBUG io.zeebe.dispatcher - Dispatcher closed
12:14:31.769 [Broker-0-LogStream-0] [-zb-actors-0] INFO  io.zeebe.logstreams - On closing logstream 0 close 1 readers
12:14:31.770 [Broker-0-LogStream-0] [-zb-actors-0] INFO  io.zeebe.logstreams - Close log storage with name 0
12:14:32.164 [Broker-0-LogStream-0] [-zb-actors-0] DEBUG io.zeebe.logstreams - Configured log appender back pressure at partition 0 as AppenderVegasCfg{initialLimit=1024, maxConcurrency=32768, alphaLimit=0.7, betaLimit=0.95}. Window limiting is disabled
12:14:32.173 [Broker-0-LogStream-0] [-zb-actors-0] INFO  io.zeebe.logstreams - Close appender for log stream 0
12:14:32.174 [0-write-buffer] [-zb-actors-4] DEBUG io.zeebe.dispatcher - Dispatcher closed
12:14:32.175 [Broker-0-LogStream-0] [-zb-actors-0] INFO  io.zeebe.logstreams - On closing logstream 0 close 1 readers
12:14:32.175 [Broker-0-LogStream-0] [-zb-actors-0] INFO  io.zeebe.logstreams - Close log storage with name 0
</pre>
</details>

**Comment by npepinpe** [comment]
I looked into it, and failed to reproduce it locally - I ran it over 3000 times, tried stressing CPU, I/O, memory, etc, anything to change the timing, but it never failed. That said, when it fails in C/I, it seems to always fail because it returns a negative value. A negative value is returned iff the dispatcher is already closed, or if it failed to claim enough space to write the full log entry. Considering this test is closing/opening, failure to claim enough space seems unlikely - maybe there's a race condition with the closed/opened condition?
